pkg/api/team_test.go

```diff
@@ -74,4 +77,53 @@ func TestTeamApiEndpoint(t *testing.T) {
 			})
 		})
 	})
+
+	t.Run("When creating team with api key", func(t *testing.T) {
+		defer bus.ClearBusHandlers()
+
+		sc := setupScenarioContext("/api/teams")
+		hs := &HTTPServer{
+			Cfg: setting.NewCfg(),
+			Bus: bus.GetBus(),
+		}
+		hs.Cfg.EditorsCanAdmin = true
+
+		teamName := "team foo"
+		sc.defaultHandler = Wrap(func(w http.ResponseWriter, c *models.ReqContext) Response {
+			c.SignedInUser = &models.SignedInUser{IsAnonymous: true}
+			c.OrgRole = models.ROLE_EDITOR
+			cmd := models.CreateTeamCommand{Name: teamName}
+			return hs.CreateTeam(c, cmd)
+		})
+		sc.m.Post(sc.url, sc.defaultHandler)
+
+		keyhash := util.EncodePassword("v5nAwpMafFP6znaS4urhdWDLS5511M42", "asd")
+		bus.AddHandler("test", func(query *models.GetApiKeyByNameQuery) error {
```

Comment:
If we get rid of the `defaultHandler` wrapper and call `CreateTeam` directly this also won't be necessary. You already override `c.SignedInUser` anyway so it is not really testing the API key middleware and if you remove the middleware change (as suggested in previous comment) it is not even necessary to test it at all.